Examples/Io/Root/src/VertexNTupleWriter.cpp

```diff
@@ -371,8 +372,10 @@ ProcessCode VertexNTupleWriter::writeT(
   // Read truth particle input collection
   const SimParticleContainer& particles = m_inputParticles(ctx);
   const SimParticleContainer& selectedParticles = m_inputSelectedParticles(ctx);
-  const TrackParticleMatching& trackParticleMatching =
-      m_inputTrackParticleMatching(ctx);
+  const TrackParticleMatching* trackParticleMatching = nullptr;
+  if (m_inputTrackParticleMatching.isInitialized()) {
+    trackParticleMatching = &m_inputTrackParticleMatching(ctx);
+  }
```

Comment:
I think a better pattern to use here is to create an empty matching map in case there is no input. This also eliminates any conditionals further down. Something similar is done here https://github.com/acts-project/acts/blob/d1a97591cba3b49dfd7a6f57d12c4a8c9675e404/Examples/Io/Root/src/RootTrackSummaryWriter.cpp#L218-L220